### PR TITLE
Add listCollaborators method

### PR DESCRIPTION
**Pull request by asoltysik**
Following https://developer.github.com/v3/repos/collaborators/#list-collaborators

### CODE REVIEW

File: github4s/shared/src/main/scala/github4s/api/Repos.scala [hunk 1]
```diff
@@ -187,6 +187,37 @@ class Repos[C, M[_]](
...
+   * organization members that are direct collaborators, organization members with access through team memberships,
+   * organization members with access through default organization permissions, and organization owners.
+   *
+   * @param accessToken to identify the authenticated user
+   * @param headers optional user headers to include in the request
+   * @param owner of the repo
+   * @param repo name of the repo
+   * @param affiliation Filter collaborators returned by their affiliation. Can be one of `outside`, `direct`, `all`.
+   *                    Default: `all`
+   * @return GHResponse[List[User]\] List of collaborators within the specified repository
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: extraneous \

**Comment by asoltysik** [comment]
The CI failing doesn't seem to by my fault?

**Comment by BenFradet** [comment]
yes only collaborators have access to the api key for the integration tests

**Comment by BenFradet** [comment]
Ran the tests locally and they're all green :+1:

**Pull request merged**